app/controllers/admin/application_controller.rb

```diff
@@ -0,0 +1,16 @@
+class Admin::ApplicationController < ApplicationController
+  before_action :authorized?
+
+  def index
+    render json: {name: "David"}.to_json
+  end
+
+  private
+
+  def authorized?
+    return if user.admin?
```

Comment:
do you have defined `user` anywhere? it will fail

Reply:
What to do with this?